### PR TITLE
Use new style Configure in the 2.0 and 2.1 templates.

### LINKED ISSUE #893: ASP.NET Core 2.0 templates set up logging in the 1.0 way
Currently our set up logging the way 1.0 used to, which still works but is not the recommended way for to add logging for 2.0.

See: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?tabs=aspnetcore2x#how-to-add-providers

### PR DESCRIPTION
**Pull request by ILMTitan**
Remove the *project.json* based templates, as we no longer support VS 2015.
Fixes #893

### CODE REVIEW

File: GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/1.1.vstemplate [hunk 1]
```diff
@@ -59,7 +59,7 @@
       <ProjectItem ReplaceParameters="true" TargetFileName="bower.json">bower.json</ProjectItem>
       <ProjectItem ReplaceParameters="false" TargetFileName=".bowerrc">.bowerrc</ProjectItem>
       <ProjectItem ReplaceParameters="true" TargetFileName="bundleconfig.json">bundleconfig.json</ProjectItem>
-      <ProjectItem ReplaceParameters="true" TargetFileName="Program.cs">Program.cs</ProjectItem>
+      <ProjectItem ReplaceParameters="true" TargetFileName="Program.cs">1.0.Program.cs</ProjectItem>
```
Comment (iantalarico) [inline GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/1.1.vstemplate]: Should this be 1.1?

File: GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/2.1.csproj [hunk 2]
```diff
@@ -14,7 +14,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Google.Cloud.Diagnostics.AspNetCore" Version="1.0.0" />
+    <PackageReference Include="Google.Cloud.Diagnostics.AspNetCore" Version="2.0.1" />
```
Comment (iantalarico) [inline GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/2.1.csproj]: Should this be 2.1.0?

File: GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/Startup.2.1.cs [hunk 3]
```diff
@@ -1,151 +0,0 @@
-﻿using Google.Cloud.Diagnostics.AspNetCore;
```
Comment (iantalarico) [inline GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/Startup.2.1.cs]: Why is this being deleted and move to the Program file?

File: GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/Program.cs [hunk 4]
```diff
@@ -1,20 +1,167 @@
...
+                        loggingBuilder.AddDebug();
+                    }
+                })
+                .Configure((app) =>
+                {
+                    var logger = app.ApplicationServices.GetService<ILoggerFactory>().CreateLogger("Startup");
+                    if (HasGcpProjectId)
+                    {
+                        // Sends logs to Stackdriver Trace.
+                        app.UseGoogleTrace();
```
Comment (iantalarico) [inline GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/Program.cs]: There's actually an even easier way to do all 3 for 2.0+ now.  Sorry I should have updated the bug

http://googlecloudplatform.github.io/google-cloud-dotnet/docs/Google.Cloud.Diagnostics.AspNetCore/

File: GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/2.1.csproj (hunk 2 above)
Comment (ILMTitan) [reply GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/2.1.csproj]: `2.0.1` is the current stable version. The first version higher than it is `3.0.0-beta01`.

Comment (iantalarico) [reply GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/2.1.csproj]: Opps yes my bad I miss read that. 2.0.1 is the one you want, which may also mean some of hte other comments aren't valid for that one

File: GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/Program.cs (hunk 4 above)
Comment (ILMTitan) [reply GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/Program.cs]: `Google.Cloud.Diagnostics.AspNetCore.GoogleDiagnosticsWebHostBuilderExtensions` does not seem to exist in *Google.Cloud.Diagnostics.AspNetCore version 2.0.1*. I would rather not depend on a beta release in the templates.

File: GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/1.1.vstemplate (hunk 1 above)
Comment (ILMTitan) [reply GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/1.1.vstemplate]: Both the 1.0 and 1.1 templates use the same file. This makes it easier to update all of the templates when we want to make a change. I may need to think a little harder about the naming convention. It should probably be either the lowest supported version (with no unversioned file) or the last/highest supported version with an unversioned file for forward versions.

File: GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/Startup.2.1.cs (hunk 3 above)
Comment (ILMTitan) [reply GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/Startup.2.1.cs]: I thought that was what the bug was about? Maybe I should have asked for more specifics.

File: GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/Program.cs (hunk 4 above)
Comment (iantalarico) [reply GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/Program.cs]: So when we init logging for 1.0 we do it like this:
```
public void Configure(IApplicationBuilder app, ILoggerFactory loggerFactory)
{
    string projectId = "[Google Cloud Platform project ID]";
    loggerFactory.AddGoogle(app.ApplicationServices, projectId);
}

```

And for 2.0 we do it like this:
```
string projectId = "[Google Cloud Platform project ID]";
var webHost = new WebHostBuilder()
    .ConfigureServices(services =>
    {
        services.AddSingleton<ILoggerProvider>(sp => GoogleLoggerProvider.Create(sp, projectId));
    })
    .UseStartup<Startup>()
    .Build();

```

Comment (iantalarico) [reply GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/Program.cs]: Sorry I should have been more clear in the

Comment (ILMTitan) [reply GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/Program.cs]: That is part of the changes that were made.

**Comment by ILMTitan** [comment]
@iantalarico PTAL.

I updated the WebAPI templates the same way. (I had forgotten to earlier).

File: GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/2.0.Program.cs [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+                })
+                .Configure((app) =>
+                {
+                    var logger = app.ApplicationServices.GetService<ILoggerFactory>().CreateLogger("Startup");
+                    if (HasGcpProjectId)
+                    {
+                        // Sends logs to Stackdriver Trace.
+                        app.UseGoogleTrace();
+                        // Sends logs to Stackdriver Error Reporting.
+                        app.UseGoogleExceptionLogging();
```
Comment (iantalarico) [inline GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/2.0.Program.cs]: Ideally this should be above trace

File: GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/2.0.Program.cs [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+        /// </returns>
+        /// <seealso href="https://cloud.google.com/error-reporting/docs/formatting-error-messages#FIELDS.service"/>
+        private static string GetServiceName(IConfiguration config)
+        {
+            var instance = Google.Api.Gax.Platform.Instance();
+            var serviceName = instance?.GaeDetails?.ServiceId ?? config["Google:ErrorReporting:ServiceName"];
+            if (string.IsNullOrEmpty(serviceName))
+            {
+                throw new InvalidOperationException(
+                    "The error reporting library needs a service name. " +
```
Comment (iantalarico) [inline GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/MVC/2.0.Program.cs]: maybe 'requires a service...'

and below

File: GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/WebAPI/2.0.Program.cs [hunk 7]
```diff
@@ -0,0 +1,170 @@
+﻿using Google.Cloud.Diagnostics.AspNetCore;
```
Comment (iantalarico) [inline GoogleCloudExtension/ProjectTemplates/GcpProjectTemplate/Core/WebAPI/2.0.Program.cs]: Same comments as in the MVC templates

**Pull request merged**